### DISCUSSION TITLE
Unable to communicate between Arduino UNO and ESP-01

### DISCUSSION THREAD

**Discussion by jimemo** [JAndrassy/WiFiEspAT#103] [General]
I have a spare Arduino Uno and an ESP-01 so I thought I would make use of them in an IOT project. 

I followed the instructions in [Getting Started](https://github.com/JAndrassy/WiFiEspAT#getting-started) and I got as far as these steps. All are saying check wiring:
* Put AT firmware a supported version of the AT firmware into the ESP you want to use with the WiFiEspAT library. Make sure the firmware is working and returning OK to test command "AT".
* If you use SoftwareSerial run the ChangeATBaudRate sketch from WiFiEspAT/Tools section in IDE Examples menu. If the sketch ends with "Communication with WiFi module failed!" check the wiring.
* Run the CheckFirmware sketch from WiFiEspAT/Tools section in IDE Examples menu. If the sketch ends with "Communication with WiFi module failed!" check the wiring.

I am using the [code examples](https://github.com/JAndrassy/WiFiEspAT/tree/master/examples) from this repo. 

These ESP-01s work with Raspberry Pico Boards and [ESP_AT_Lib](https://github.com/khoih-prog/ESP_AT_Lib) using hardware serial baud rate 115200 but I read this:
_Over hardware Serial this library works reliably with AT firmware's default 115200 baud. With SoftwareSerial, 9600 baud should be used because of limitations of SoftwareSerial._

**Question -** I would rather not change the baud rates of these chips, shall I use hardware TX/RX instead? If so:
* What pins to I connect TX/RX on the Uno?
* Do I need to add this line at the top of my sketch or somewhere else? `#define HAVE_HWSERIAL1 true`

Please can somebody spot what I am doing wrong?

ESP-01 Firmware:
```
FW Version: AT+GMR
AT version:2.2.0.0(b097cdf - ESP8266 - Jun 17 2021 12:57:45)
SDK version:v3.4-22-g967752e2
compile time(6800286):Aug  4 2021 17:20:05
Bin version:2.2.0(Cytron_ESP-01S)
```

Wiring (both EN & 3V3 on the ESP-01 are connected to single 3V3 on the Uno):
| Uno | ESP-01 |
| ------------- | ------------- |
| D7 | RX |
| D6 | TX |
| 3V3 |EN|
| 3V3 | 3V3|
| GND |GND |

**Comment by JAndrassy** [comment]
Classic Nano has only one hardware Serial, which is connected to the USB chip to communicate with the computer (sketch upload, Serial Monitor). the RX and TX pins are the same Serial so if you wire the esp-01 to RX/TX you will have problems to upload a sketch to Nano and debug it.

The 3.3 V pin on Nano can't provide enough current for the esp-01 to do WiFi transmit. You need an external 3.3 V power supply for the esp-01.
